cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -2277,16 +2277,16 @@ Slice::JavaGenerator::getTypeMetadata(const MetadataList& metadata, string& inst
     {
         if (m->directive() == "java:type")
         {
-            string_view arguments = m->arguments();
+            string arguments = m->arguments();
             string::size_type pos = arguments.find(':');
             if (pos != string::npos)
             {
-                instanceType = string{arguments.substr(0, pos)};
-                formalType = string{arguments.substr(pos + 1)};
+                instanceType = arguments.substr(0, pos);
+                formalType = arguments.substr(pos + 1);
             }
             else
             {
-                instanceType = string{arguments};
+                instanceType = arguments;
```

Comment:
If performance was a concern here (it's not), you could write:
```cpp
instanceType = std::move(arguments);
```

There is even a `&&` version of substr as of C++23 - see https://en.cppreference.com/w/cpp/string/basic_string/substr.

Reply:
Still, there's no reason not to move here, so I did!